libs/order-list/src/lib/components/order-list/order-list.tsx

```diff
@@ -1,24 +1,78 @@
 import { OrderTimeInForce, OrderStatus, Side } from '@vegaprotocol/types';
-import type { Orders_party_orders } from './__generated__/Orders';
+import type { Orders_party_orders } from '../__generated__/Orders';
 import {
   addDecimal,
-  formatNumber,
+  addDecimalsFormatNumber,
   getDateTimeFormat,
   t,
 } from '@vegaprotocol/react-helpers';
-import { AgGridDynamic as AgGrid } from '@vegaprotocol/ui-toolkit';
-import type { ValueFormatterParams } from 'ag-grid-community';
+import { AgGridDynamic as AgGrid, Button } from '@vegaprotocol/ui-toolkit';
+import type {
+  ICellRendererParams,
+  ValueFormatterParams,
+} from 'ag-grid-community';
 import type { AgGridReact } from 'ag-grid-react';
 import { AgGridColumn } from 'ag-grid-react';
-import { forwardRef } from 'react';
+import { forwardRef, useState } from 'react';
+import { useOrderCancel } from '@vegaprotocol/wallet';
+import { CancelDialog } from '../cancel-order-dialog/cancel-dialog';
 import BigNumber from 'bignumber.js';
 
 interface OrderListProps {
   data: Orders_party_orders[] | null;
 }
+interface OrderListTableProps {
+  data: Orders_party_orders[] | null;
+  cancel: (body?: unknown) => Promise<unknown>;
+}
 
 export const OrderList = forwardRef<AgGridReact, OrderListProps>(
   ({ data }, ref) => {
+    const [orderDialogOpen, setOrderDialogOpen] = useState(false);
+    const { transaction, finalizedOrder, reset, cancel } = useOrderCancel();
+    return (
+      <>
+        <OrderListTable data={data} cancel={cancel} ref={ref} />
+        <CancelDialog
+          {...{
+            orderDialogOpen,
+            setOrderDialogOpen,
+            finalizedOrder,
+            transaction,
+            reset,
+          }}
+        />
+      </>
+    );
+  }
+);
+
+export const OrderListTable = forwardRef<AgGridReact, OrderListTableProps>(
+  ({ data, cancel }, ref) => {
+    const CancelRendererButton = ({ data }: ICellRendererParams) => {
```

Comment:
chore (blocking): I don't see benefit of creating a function here its not used mored than once and just results in you having to scroll up again once you get to this cell in the render logic below.  Can we move it inline?
```
cellRenderer={({ data} : ICellRendererPrarams) => ...}
```